p2panda-rs/src/operation/operation_action.rs

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::fmt;
+use std::fmt::Display;
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Operations are categorised by their action type.
+///
+/// An action defines the operation format and if this operation creates, updates or deletes a data
+/// document.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum OperationAction {
```

Comment:
In the spec this is called _operation type_, which is a bit more generic. Would you mind if I rename it?

Reply:
Oh! I have the feeling we've been always talking about "operation action", so its hard for me to think about anything else. Its a relatively big change to change it to "type" now as many other places use the word "action" .. what do you think @sandreae?

Reply:
Hmm, I'm scared to introduce more renaming in the code right now... :grimacing:

Reply:
I can understand that :)